lighthouse/lighthouse-auth.mjs

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable no-console */
+
+import fs from 'fs';
+
+/**
+ * @param {puppeteer.Page} page
+ * @param {string} origin
+ */
+async function login(page, origin) {
+  if (process.env.PREVIEW_URL) {
+    await loadInitialPage(page, origin);
+  } else {
+    await page.goto(origin);
+  }
+
+  let signInButton;
+  try {
+    signInButton = await page.$('#sign-in-button');
+    if (signInButton === null) {
+      return;
+    }
+  } catch (error) {
+    console.log(' Did not find signInButton', error);
+    return;
+  }
+
+  // Redirect to Okta login
+  const redirectToOktaPromise = page.waitForNavigation();
+  await signInButton.click();
+  await redirectToOktaPromise;
+
+  // Fill in and submit login form.
+  try {
+    await loginToOkta(page);
+  } catch (error) {
+    // Sometimes the login to Okta fails for some reason and puts us back on the /login page of MPDx, so try again.
+    try {
+      signInButton = await page.$('#sign-in-button', { timeout: 5000 });
+
+      if (signInButton) {
+        const redirectToOktaPromise = page.waitForNavigation();
+        await signInButton.click();
+        await redirectToOktaPromise;
+        try {
+          await loginToOkta(page);
+        } catch (loginError) {
+          console.log('Failed to login', loginError);
+        }
+      }
+    } catch (error) {
+      console.log(' Did not find signInButton', error);
+      return;
+    }
+  }
+}
+
+export async function loadInitialPage(page, origin, timeout = 60_000) {
+  let originResponse = await page.goto(origin);
+
+  return await new Promise((resolve) => {
```

Comment:
I think this will wait `timeout` even if the page succeeds the first time. Maybe you could add something like this:

```ts
if (originResponse.status && originResponse.status !== 404) {
  resolve(originResponse);
  return;
}
```